drivers/Aqara/aqara-lock/src/init.lua

```diff
@@ -0,0 +1,233 @@
+  device:emit_event(TamperAlert.tamper("clear", { visibility = { displayed = false } }))
+  device:emit_event(Lock.lock("locked"))
+  credential_utils.save_data(self)
+end
+
+local function toValue(payload, start, length)
+  local ret = 0
+  for i = start, start + length - 1 do
+    ret = (ret << 8) + string.byte(payload, i)
+  end
+  return ret
+end
+
+local function toHex(value, length)
+  local ret = string.char(0xFF & value)
+  for i = length, 2, -1 do
+    ret = string.char(0xFF & (value >> 8 * (i - 1))) .. ret
+  end
+  return ret
+end
+
+local function event_lock_handler(driver, device, evt_name, evt_value)
+  if evt_value == 1 then
+    device:emit_event(Lock.lock(evt_name))
+    device:emit_event(TamperAlert.tamper.clear())
+    if credential_utils.is_exist_host(device) then
+      device:emit_event(remoteControlStatus.remoteControlEnabled('true', { visibility = { displayed = false } }))
+    else
+      device:emit_event(remoteControlStatus.remoteControlEnabled('false', { visibility = { displayed = false } }))
+    end
+  end
+end
+
+local function event_unlock_handler(driver, device, evt_name, evt_value)
+  local id, label
+  id, label = credential_utils.find_userLabel(driver, device, evt_value)
+  device:emit_event(Lock.lock.unlocked({ data = { method = evt_name, codeId = id, codeName = label } }))
+  device:emit_event(remoteControlStatus.remoteControlEnabled('false', { visibility = { displayed = false } }))
+  device:emit_event(TamperAlert.tamper.clear())
+end
+
+local function event_door_handler(driver, device, evt_name, evt_value)
+  if evt_value == 2 then
+    device:emit_event(Lock.lock(evt_name))
+    device:emit_event(TamperAlert.tamper.clear())
+  elseif evt_value == 4 then
+    device:emit_event(TamperAlert.tamper.detected())
+  end
+end
+
+local function event_battery_handler(driver, device, evt_name, evt_value)
+  device:emit_event(Battery.battery(evt_value))
+end
+
+local function event_temper_alert_handler(driver, device, evt_name, evt_value)
+  device:emit_event(TamperAlert.tamper.detected())
+end
+
+local resource_id = {
+  { id = "13.31.85", event_name = "locked",       event_handler = event_lock_handler },
+  { id = "13.17.85", event_name = "not fully locked", event_handler = event_door_handler },
+  { id = "13.48.85", event_name = "manual",       event_handler = event_unlock_handler },
+  { id = "13.51.85", event_name = "manual",       event_handler = event_unlock_handler },
+  { id = "13.42.85", event_name = "fingerprint",  event_handler = event_unlock_handler },
+  { id = "13.43.85", event_name = "keypad",       event_handler = event_unlock_handler },
+  { id = "13.44.85", event_name = "rfid",         event_handler = event_unlock_handler },
+  { id = "13.45.85", event_name = "bluetooth",    event_handler = event_unlock_handler },
+  { id = "13.90.85", event_name = "command",      event_handler = event_unlock_handler },
+  { id = "13.46.85", event_name = "keypad",       event_handler = event_unlock_handler },
```

Comment:
for these method strings, we generally make a little "enum" to keep track of these strings: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1298

Reply:
Okay. I modified it.